### ISSUE TITLE
setFieldValue doesn't run field level validations

### ISSUE THREAD

**Issue by Andreyco** [jaredpalmer/formik#734]
## Current Behavior
Currently, when using `formik.setFieldValue` with `shouldValidate` flag set to `true` field level validations are not run.

## Steps to Reproduce
- Define field level validation function. eg. `<Field name="name" validate={() => console.log('validate')} />`
- Run `formik.setFieldValue('name', 'Andrej', true)` and see console for output. Nothing s will show.

## Expected behavior
`validate` should appear in console logs

## Suggested solution(s)
Modify `setFieldValue` handler to run validations on registered fields.

## Additional context

**CodeSandbox Link**:
https://codesandbox.io/s/n57146642l

---

- Formik Version: 1.0.0-beta.4
- React Version: 15.6.1

**Comment by jaredpalmer** [comment]
Great idea! Shouldn’t be too hard to add. We should also do it on setFieldTouched too.


—-

While we are here, we can actually do even more with the latest refactor.....we can let you pick which field validation function(s) to run. I haven’t yet experimented too much with this idea yet, but am open to suggestions. I think a validateField function could be very useful.

**Comment by Andreyco** [comment]
Agree, setFieldTouched should run them as well. Could you point me to refactored code so that I can take a look?

**Comment by jaredpalmer** [comment]
I’m on mobile. Take a look at executeSubmit, this.fields, and registerField and unregisterField

**Comment by Andreyco** [comment]
So far, I found it would be great to run field level validation from these handlers
- setFieldValue
- setFieldTouched
- setTouched
- setValues

Which boils down to having support for field level validation in runValidations handler.

**Comment by Andreyco** [comment]
@jaredpalmer I attempted a refactor for validation handlers. Would like to have it reviewed ;)

**Comment by siddhant-rl** [comment]
Just need to add `await` in front of `setFieldValue`

``` javascript
await setFieldValue(`key`, value);
```
